### ISSUE TITLE
DROP CLUSTER does not send DropInstance to computed

### ISSUE THREAD

**Issue by lluki** [MaterializeInc/materialize#14687]
### What version of Materialize are you using?

v0.26.1-dev (4e24f4b6a)

### How did you install Materialize?

Built from source

### What is the issue?

Creating and dropping a cluster does not correctly shut down timely on the computeds. The `DropInstance` command gets swallowed somewhere on the way.

The issue can be reproduced by adding a print statement in `src/compute/src/server.rs:446` and executing
```
> CREATE CLUSTER c1 REPLICAS (r1 (SIZE '1'));
> DROP CLUSTER c1;
```

It potentially can keep consuming resources, although i suspect in cloud this is not an issue as we also remove the compute pod from kubernetes.

**Comment by uce** [comment]
This bug has been moved to the backlog of compute. We don't need the P3 label for bugs, so I'm removing.

**Comment by lluki** [comment]
This was fixed in #15467
